### DISCUSSION TITLE
Question on why the model that i train cannot predict my test set.

### DISCUSSION THREAD

**Discussion by keithajy** [roboflow/supervision#325] [Q&A]
### Search before asking

- [X] I have searched the Supervision [issues](https://github.com/roboflow/supervision/issues) and found no similar feature requests.

### Question

How do I train my data if i have a small amount of data set but, sometimes it works for the images i train, sometimes it doesnt, is the error in my images that i am feeding in to train.

**Comment by github-actions** [comment]
Hello there, thank you for opening an Issue ! 🙏🏻 The team was notified and they will get back to you asap.

**Comment by keithajy** [comment]
i am new to this, i am looking to train images that contains logo to detect logos on documents. Some logos works out better than the others, am not sure if its the logo problem or just me not passing in the right image data.

**Comment by hardikdava** [comment]
@keithajy , it seems like a more of the model training issue. Supervision does not support feature of model training. Once you have model then you can use the supervision for annotating model predictions, model benchmarking as well as some counting application. But training of a model does not in a scope of supervision. Feel free to raise an issue regarding supervision. For more information, please check out https://github.com/roboflow/notebooks to see various options to train a model.

**Comment by keithajy** [comment]
@hardikdava does this mean that i have to use a pretrain model from roboflow/notebooks? or that I have to raise a question there instead? thank you so much!

**Comment by hardikdava** [comment]
@keithajy roboflow/notebooks provides options and guide on how to train a model with various frameworks like yolov8, yolo-nas, etc. Your steps would be like as following.

- Choose a framework to train model
- Use roboflow to annotate your data
- Export in required format i.e. coco, yolo, etc
- Train a model with guidance provided by roboflow/notebooks,
- Use supervision afterwards.

I am afraid we can not help you anymore on this issue but feel free to open an issue on framework repo you choose for model training.

**Comment by SkalskiP** [comment]
Hi, @hardikdava 👋🏻! Thanks a lot for your help and all the suggestions. @keithajy I hope you found that helpful. If you don't mind, I'll convert that issue into a discussion and nice it to the Q/A section.
